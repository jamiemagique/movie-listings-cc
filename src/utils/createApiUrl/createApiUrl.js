import { API_BASEURL, API_KEY } from '../../config';

/**
 * Creates a URL to the API.
 *
 * @param {string} path - API resource, including a `/` prefix.
 * @returns {string} - API url
 */
const createApiUrl = path => `${API_BASEURL}${path}?api_key=${API_KEY}`;

export default createApiUrl;
