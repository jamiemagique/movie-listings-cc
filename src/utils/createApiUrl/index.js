import createApiUrl from './createApiUrl';

export default createApiUrl;
