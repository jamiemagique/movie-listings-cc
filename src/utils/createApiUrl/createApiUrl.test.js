import createApiUrl from '.';

/**
 * We mock the configuration values so if they 
 * change over time our tests don't fail, also
 * we don't need to set up any env variables.
 */
jest.mock('../../config', () => ({
  API_BASEURL: 'api.example.com',
  API_KEY: '<api_key>',
}));

it('creates an api url correctly', () => {
  expect(createApiUrl('/configuration')).toMatchSnapshot();
});
