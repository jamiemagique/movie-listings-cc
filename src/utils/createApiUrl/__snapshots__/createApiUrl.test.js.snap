// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`creates an api url correctly 1`] = `"api.example.com/configuration?api_key=<api_key>"`;
